packages/desktop-client/src/components/budget/index.js

```diff
@@ -29,38 +36,45 @@ import { ReportProvider } from './report/ReportContext';
 import * as rollover from './rollover/rollover-components';
 import { RolloverContext } from './rollover/RolloverContext';
 
-let _initialBudgetMonth = null;
-
-class Budget extends PureComponent {
-  constructor(props) {
-    super(props);
-
-    const currentMonth = _initialBudgetMonth || monthUtils.currentMonth();
-    this.state = {
-      initialized: false,
-      prewarmStartMonth: props.startMonth || currentMonth,
-      newCategoryForGroup: null,
-      isAddingGroup: false,
-      collapsed: props.collapsedPrefs || [],
-      bounds: { start: currentMonth, end: currentMonth },
-      categoryGroups: null,
-      summaryCollapsed: props.summaryCollapsed,
-    };
-  }
+const currentMonth = monthUtils.currentMonth();
+
+function Budget(props) {
+  const tableRef = useRef(null);
+
+  const [initialized, setInitialized] = useState(false);
+  const [prewarmStartMonth, setPrewarmStartMonth] = useState(
+    props.startMonth || currentMonth,
+  );
+  const [newCategoryForGroup, setNewCategoryForGroup] = useState(null);
+  const [isAddingGroup, setIsAddingGroup] = useState(false);
+  const [collapsed, setCollapsed] = useState(props.collapsedPrefs || []);
+  const [bounds, setBounds] = useState({
+    start: currentMonth,
+    end: currentMonth,
+  });
+  const [categoryGroups, setCategoryGroups] = useState(null);
+  const [summaryCollapsed, setSummaryCollapsed] = useState(
+    props.summaryCollapsed,
+  );
 
-  async loadCategories() {
-    let result = await this.props.getCategories();
-    this.setState({ categoryGroups: result.grouped });
+  async function loadCategories() {
+    let result = await props.getCategories();
+    setCategoryGroups(result.grouped);
   }
 
-  async componentDidMount() {
-    let { titlebar, budgetType } = this.props;
-    this.loadCategories();
+  useEffect(() => {
+    let { titlebar, budgetType } = props;
 
-    let { start, end } = await send('get-budget-bounds');
-    this.setState({ bounds: { start, end } });
+    async function run() {
+      loadCategories();
 
-    this.prewarmAllMonths({ start, end }, budgetType);
+      let { start, end } = await send('get-budget-bounds');
+      setBounds({ start, end });
+
+      prewarmAllMonths({ start, end }, budgetType);
+    }
+
+    run();
```

Comment:
I think this will behave differently than `componentDidMount` as it will silently swallow exceptions where before it will bubble up. Do we want to add a `.catch(console.error)` here just in case there's as error?

Reply:
Just tested locally by adding a `throw new Error();`: both cases (current & before) behave exactly the same. Adding `catch(console.error)` also does _basically_ the same (as in: just logs the error, but in `catch` case the error is not reported as "unhandled").

So TLDR: even though error handling in this component is pretty shaky.. there is no regression from current implementation.